renku/api/repository.py

```diff
@@ -216,8 +216,12 @@ def is_cwl(self, path):
         """Check if the path is a valid CWL file."""
         return path.startswith(self.cwl_prefix) and path.endswith('.cwl')
 
-    def find_previous_commit(self, paths, revision='HEAD'):
-        """Return a previous commit for a given path."""
+    def find_previous_commit(self, paths, revision='HEAD', first=False):
+        """Return a previous commit for a given path starting from ``revision``.
+
+        :param revision: revision to start from, defaults to ``HEAD``
+        :param first: instead of last, show the first commit in the history
```

Comment:
If we are doing this change then let's add the most important part of it: exceptions being raised =))

Reply:
done